adapter/internal/oasparser/model/open_api_31.go

```diff
@@ -0,0 +1,265 @@
+	// If it's present, then the same value should be added to the
+	// resource level if vendor ext is not present at each resource level.
+	val, found := resolveDisableSecurity31(openAPI.Extensions)
+	if openAPI.Paths != nil {
+		conf, _ := config.ReadConfigs()
+		for path, pathItem := range openAPI.Paths.PathItems.FromOldest() {
+			if conf.Envoy.MaximumResourcePathLengthInKB != -1 &&
+				isResourcePathLimitExceeds(path, int(conf.Envoy.MaximumResourcePathLengthInKB)) {
+				return nil, errors.New("path: " + path + " exceeds maximum allowed length")
+			}
+			// Checks for resource level security. (security is disabled in resource level using x-wso2-disable-security extension)
+			isResourceLvlSecurityDisabled, foundInResourceLevel := resolveDisableSecurity31(pathItem.Extensions)
+			methodsArray := make([]*Operation, pathItem.GetOperations().Len())
+			var arrayIndex int = 0
+			for httpMethod, operation := range pathItem.GetOperations().FromOldest() {
+				if operation != nil {
+					if foundInResourceLevel {
+						operation.Extensions = addDisableSecurityIfNotPresent31(operation.Extensions, isResourceLvlSecurityDisabled)
+					} else if found {
+						operation.Extensions = addDisableSecurityIfNotPresent31(operation.Extensions, val)
+					}
+					methodsArray[arrayIndex] = getOperationLevelDetails31(operation, strings.ToUpper(httpMethod))
+					arrayIndex++
+				}
+			}
+
+			resource := setPathInfoOpenAPI31(path, methodsArray, pathItem)
+			var productionUrls []Endpoint
+			if isServerURLIsAvailable31(pathItem.Servers) {
+				for _, serverEntry := range pathItem.Servers {
+					if len(serverEntry.URL) == 0 || strings.HasPrefix(serverEntry.URL, "/") {
+						continue
+					}
+					endpoint, err := getHTTPEndpoint(serverEntry.URL)
+					if err == nil {
+						productionUrls = append(productionUrls, *endpoint)
+
+					} else {
+						logger.LoggerOasparser.Info("Not considering the URL in servers object as parsing has failed. ", err)
+					}
+
+				}
+				if len(productionUrls) > 0 {
+					resource.productionEndpoints = generateEndpointCluster(prodClustersConfigNamePrefix, productionUrls, LoadBalance)
+				}
+			}
+			resources = append(resources, &resource)
+
+		}
+	}
+	return SortResources(resources), nil
+}
+
+// setSecuritySchemesOpenAPI31 extracts the security schemes from the openAPI v3.1 definition and returns a slice of SecurityScheme structs.
+func setSecuritySchemesOpenAPI31(openAPI v3.Document) []SecurityScheme {
+	var securitySchemes []SecurityScheme
+	for key, val := range openAPI.Components.SecuritySchemes.FromOldest() {
+		scheme := SecurityScheme{DefinitionName: key, Type: val.Type, Name: val.Name, In: val.In}
+		securitySchemes = append(securitySchemes, scheme)
+	}
+	logger.LoggerOasparser.Debugf("Security schemes in setSecuritySchemesOpenAPI method %v:", securitySchemes)
+	return securitySchemes
+}
+
+func getOperationLevelDetails31(operation *v3.Operation, method string) *Operation {
```

Comment:
```suggestion
func getOperationLevelDetailsForOAS31(operation *v3.Operation, method string) *Operation {
```
Check for other function names as well.

Reply:
Changed